circuit_knitting/cutting/cutting_reconstruction.py

```diff
@@ -79,22 +100,33 @@ def reconstruct_expectation_values(
         label: ObservableCollection(subobservables)
         for label, subobservables in subobservables_by_subsystem.items()
     }
+    sorted_subsystems = sorted(subsystem_observables.keys())
 
-    # Assign each weight's sign and calculate the expectation values for each observable
-    for i, coeff in enumerate(coefficients):
-        sorted_subsystems = sorted(subsystem_observables.keys())  # type: ignore
+    # Reconstruct the expectation values
+    for i in range(len(coefficients)):
         current_expvals = np.ones((len(expvals),))
-        for j, label in enumerate(sorted_subsystems):
+        for label in sorted_subsystems:
             so = subsystem_observables[label]
+            coeff = coefficients[i]
             subsystem_expvals = [
                 np.zeros(len(cog.commuting_observables)) for cog in so.groups
             ]
             for k, cog in enumerate(so.groups):
-                quasi_probs = quasi_dists[i][j][k][0]
+                quasi_probs = results_dict[label].quasi_dists[i * len(so.groups) + k]
                 for outcome, quasi_prob in quasi_probs.items():
-                    subsystem_expvals[k] += quasi_prob * _process_outcome(
-                        quasi_dists[i][j][k][1], cog, outcome
-                    )
+                    try:
+                        subsystem_expvals[k] += quasi_prob * _process_outcome(
+                            results_dict[label].metadata[i * len(so.groups) + k][
+                                "num_qpd_bits"
+                            ],
+                            cog,
+                            outcome,
+                        )
+                    except KeyError:
+                        raise ValueError(
+                            "The num_qpd_bits field must be set in each subexperiment "
+                            "result metadata dictionary."
+                        )
```

Comment:
My goal here is to have the _minimal_ amount of code in the `try` block, so it can be surgically precise.  If, hypothetically, `_process_outcome` were to raise some unrelated `KeyError`, we don't want to catch that here.  (Not that it will, but we only know that from nonlocal reasoning, which is better avoided.)

Also: I bet black will want to reformat this.

```suggestion
                    try:
                        num_qpd_bits = results_dict[label].metadata[i * len(so.groups) + k][
                            "num_qpd_bits"
                        ]
                    except KeyError:
                        raise ValueError(
                            "The num_qpd_bits field must be set in each subexperiment "
                            "result metadata dictionary."
                        )
                    else:
                        subsystem_expvals[k] += quasi_prob * _process_outcome(
                            num_qpd_bits,
                            cog,
                            outcome,
                        )
```